frontend/src/components/Conversation/Conversation.tsx

```diff
@@ -35,11 +41,20 @@ export const Conversation = () => {
   const { data: connectionsData } = useGetConnections();
   const { data: conversationsData } = useGetConversations();
 
+  const [tempResults, setTempResults] = useState<IResultType[]>([]);
+
   const {
     mutate: sendMessageMutation,
     isPending: isPendingSendMessage,
     variables: newMessageVariable,
-  } = useSendMessage();
+  } = useSendMessageStreaming({
+    onAddResult: (result) =>
+      setTempResults((prev) => [
+        ...prev,
+        { ...result, result_id: generateUUID() },
```

Comment:
How does this play with things like chart updating/query updating? Cause for those we'd need the actual result ID right?

Reply:
These are temporary, not meant to be modified. Once the stream ends the `tempResults` state is emptied (`setTempResults([])`), and the mutation's `onSuccess` properly modifies the cached messages with results that have proper result ids and linked ids

Reply:
Ok I understand